custom_components/gtfs_rt/sensor.py

```diff
@@ -134,6 +138,12 @@ def extra_state_attributes(self):
             attrs[ATTR_NEXT_UP] = next_buses[1].arrival_time.strftime(TIME_STR_FORMAT) if len(next_buses) > 1 else '-'
             attrs[ATTR_NEXT_UP_DUE_IN] = due_in_minutes(next_buses[1].arrival_time) if len(next_buses) > 1 else '-'
             attrs[ATTR_NEXT_OCCUPANCY] = next_buses[1].occupancy
+        if len(next_buses) > 2 and (self._max_departures > 2 or self._max_departures == 0):
+            end_index = min(len(next_buses), self._max_departures + 2) if self._max_departures > 0 else len(next_buses)
```

Comment:
With the convention that `max_departures == -1` means to get all buses, you can do `end_index = None if max_departures == -1 else max_departures`. `None` will make sure you get the whole list and if `len(next_buses) < max_departures` then `next_buses[2:max_departures]` is still safe in Python (it will return a truncated list).